frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideInNewBranchBelowAction.java

```diff
@@ -0,0 +1,136 @@
+      presentation.setDescription(GitMacheteBundle.message("action.slide-in.description.disabled.no-parent"));
+    } else {
+      // it is supposed to be defined since "branch" is defined
+      assert branchName.isDefined() : "Branch name is undefined";
+      presentation.setDescription(GitMacheteBundle.message("action.slide-in.description", branchName.get()));
+
+      if (getCurrentBranchNameIfManaged(anActionEvent).equals(branchName)) {
+        presentation.setText(GitMacheteBundle.message("action.slide-in.text"));
+      }
+    }
+  }
+
+  @Override
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    var project = getProject(anActionEvent);
+    var selectedVcsRepository = getSelectedGitRepository(anActionEvent);
+    var gitMacheteParentBranch = getNameOfBranchUnderAction(anActionEvent)
+        .flatMap(bn -> getGitMacheteBranchByName(anActionEvent, bn));
+    var hash = gitMacheteParentBranch.map(b -> b.getPointedCommit()).map(c -> c.getHash());
+    var branchLayout = getBranchLayout(anActionEvent);
+    var branchLayoutWriter = getBranchLayoutWriter(anActionEvent);
+
+    if (selectedVcsRepository.isEmpty() || gitMacheteParentBranch.isEmpty() || branchLayout.isEmpty()) {
+      return;
+    }
+
+    var branchName = createOrCheckoutNewBranch(project, selectedVcsRepository.get(), gitMacheteParentBranch.get().getName(),
+        GitMacheteBundle.message("action.slide-in.dialog.title"));
+    if (branchName == null) {
+      log().debug("Name of branch to slide in is null: most likely the action has been canceled from dialog");
+      return;
+    }
+
+    new Task.Backgroundable(project, GitMacheteBundle.message("action.slide-in.task.title")) {
+
+      @Override
+      public void run(ProgressIndicator indicator) {
+        Path macheteFilePath = getMacheteFilePath(selectedVcsRepository.get());
+        var notifier = VcsNotifier.getInstance(project);
+
+        var newBranchLayout = Try.of(() -> branchLayout.get().slideIn(gitMacheteParentBranch.get().getName(), branchName))
+            .onFailure(
+                t -> notifier.notifyError(/* title */ GitMacheteBundle.message("action.slide-in.notification.fail", branchName),
+                    getMessageOrEmpty(t)))
+            .toOption();
+
+        newBranchLayout.map(nbl -> Try.of(() -> {
+          branchLayoutWriter.write(macheteFilePath, nbl, /* backupOldLayout */ true);
+          return "ok";
+        }).onFailure(t -> notifier.notifyError(/* title */ GitMacheteBundle.message("branch-layout.write.notification.fail"),
+            getMessageOrEmpty(t))));
+      }
+    }.queue();
+  }
+
+  private static String getMessageOrEmpty(Throwable t) {
+    return t.getMessage() != null ? t.getMessage() : "";
+  }
+
+  @Nullable
+  String createOrCheckoutNewBranch(Project project, GitRepository gitRepository, String startPoint, String title) {
+    var repositories = List.of(gitRepository).asJava();
+    var options = new GitNewBranchDialog(project,
```

Comment:
Whoops internal API... does it seem to be stable...?

Reply:
Also, the problem with using a **new branch** dialog here is that we don't allow for sliding in an existing branch, which is kinda useful... although this feature can be provided on a separate PR

Reply:
(or even under a seperate issue... coz we might need to postpone that into v0.7.0)

Reply:
(unless it's easy to add to the existing impl)

Reply:
whoops. probably a very accurate point about the stability :confused: 

It is likely that another checkbox `setTrackingCheckbox` will be added in the near future.
So the next custom dialog is required... :rocket:

Reply:
> Also, the problem with using a **new branch** dialog here is that we don't allow for sliding in an existing branch, which is kinda useful... although this feature can be provided on a separate PR

What do we mean by `sliding in an existing branch`? What's to be existing? Name within the branch layout? A local branch absent in there?

Reply:
Probably both cases shall be handled :thinking:

Reply:
I specifically meant a machete-unmanaged local branch

Reply:
> Whoops internal API... does it seem to be stable...?

Tested on 202.* EAP. Works fine because the new property has the default value set. I think we can leave it like this (without custom dialog impl)

![image](https://user-images.githubusercontent.com/19799111/87060781-36eaae80-c20b-11ea-832d-8b62c9d60087.png)